NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h

```diff
@@ -299,6 +325,11 @@ class NEOMATHENGINE_API IBlasEngine : public IVectorMathEngine {
 	virtual void AddMatrixElementsToVector(const CConstFloatHandle& matrix, int height, int width,
 		const CConstIntHandle& rowIndices, const CConstIntHandle& columnIndices,
 		const CFloatHandle& result, int vectorSize) = 0;
+
+	// Elementwise adds two matrices of the same size
+	virtual void AddDiagMatrixToMatrix( const CConstFloatHandle& diagMatrix, const CConstFloatHandle& matrix,
```

Comment:
What for both `height` and `width` are required here? It's supposed that second matrix is a square.

In current realization we can add squared matrix to the non-squared, but that doesn't make any sense.